volume/store/store.go

```diff
@@ -23,14 +23,24 @@ const (
 )
 
 type volumeMetadata struct {
-	Name   string
-	Labels map[string]string
+	Name    string
+	Labels  map[string]string
+	Options map[string]string
 }
 
 type volumeWrapper struct {
 	volume.Volume
-	labels map[string]string
-	scope  string
+	labels  map[string]string
+	scope   string
+	options map[string]string
+}
+
+func (v volumeWrapper) Options() map[string]string {
+	options := map[string]string{}
+	for key, value := range v.options {
+		options[key] = value
+	}
+	return options
```

Comment:
Thanks @yongtang. I know that deep copy is correct. While I still can not follow why options needs a deep copy, while labels does not. @cpuguy83 Can you give me some explaination. Thanks a lot.